### ISSUE TITLE
No ability to set manual limits for paragraph block's “Custom Size” option slider

### ISSUE THREAD

**Issue by josephdickson** [WordPress/gutenberg#8369]
Hi Everyone,

Currently as of Version 3.4.0 a user can slide the **Custom Size** slider from 12px to 100px and there appears to be no documented way limit or change this range for theme or plugin developers.

![paragrpah custom font size slider limit at 48px](https://user-images.githubusercontent.com/9701927/43534057-43314024-956b-11e8-8564-2510e3dae001.PNG)

It is useful to give some granular control to content creators but there is no practical reason why a paragraph should be increased to 100px on an informational page.

Desktop Example:
![image](https://user-images.githubusercontent.com/9701927/43534367-0ad4b250-956c-11e8-9f11-48fd6fbf0d96.png)

Mobile Example:
![image](https://user-images.githubusercontent.com/9701927/43536107-9bb4098e-9570-11e8-80df-a554b212f92b.png)

**Ideal Solution**
Gutenberg's core blocks should allow for plugin and theme customization of all their individual features.

Just as we can create custom settings for the [Small, Medium, Large](https://wordpress.org/gutenberg/handbook/extensibility/theme-support/#block-font-sizes) buttons we should be able to manually set the font-size range from 16px - 32px; and go beyond 100px or below 12px.

Ideally we should also be able to **customize or disable** any sub component without blacklisting the entire block. Whether it's the preset buttons, custom font size slider or setting a drop cap.

**Current Alternative**
I could [blacklist](https://wordpress.org/gutenberg/handbook/extensibility/extending-blocks/#using-a-blacklist) the core paragraph block and create my own replacement.

See forum support ticket for additional details at [https://wordpress.org/support/topic/how-to-limit-paragraph-custom-size-options/](https://wordpress.org/support/topic/how-to-limit-paragraph-custom-size-options/)

**Comment by josephdickson** [comment]
Let me know if I be of any assistance in clarifying this request.
Thanks,
Joe

**Comment by katerlouis** [comment]
What's the status here? 
I'm very interested in limiting the paragraph options aswell.
Is this even planned? In my opinion restricting choices the editor can make is crucial for a cohesive design.

**Comment by chrisvanpatten** [comment]
This is a case where I would suggest disabling support for custom font sizes and registering your own set. I don’t think it’s likely that we would support limits on the custom option, at least not until phase 2.

**Comment by josephdickson** [comment]
Still open.
@chrisvanpatten is there documentation for this sort of thing the handbook at https://wordpress.org/gutenberg/handbook/extensibility/extending-blocks/ doesn't cover it.

**Comment by ajitbohra** [comment]
@katerlouis @josephdickson 

Docs for disabling custom font and registering your own set is under `extensibility > Theme Support` https://wordpress.org/gutenberg/handbook/extensibility/theme-support/#block-font-sizes

**Comment by josephdickson** [comment]
@ajitbohra, My mistake, I hadn't tested for 4.0
Looks like the Range slider was removed in favor of the picker. Can anyone else confirm?
[Example Screenshot of Picker](https://user-images.githubusercontent.com/9701927/47183788-ca64a900-d2dd-11e8-9286-ac27b113dd69.png)
)

**Comment by chrisvanpatten** [comment]
@josephdickson Yes the picker is new for 4.0. And the input field for the number will be hideable with `add_theme_support( 'disable-custom-font-sizes' );` once Gutenberg 4.1 is released.

**Comment by josephdickson** [comment]
@chrisvanpatten fantastic! forward my thanks to everyone involved in adjusting this feature.

I guess this is closed once 4.1 comes out :)

**Comment by josephdickson** [comment]
I've decided to re-open, Should this be considered resolved?

**Comment by designsimply** [comment]
@josephdickson have you tried the solution proposed at https://github.com/WordPress/gutenberg/issues/8369#issuecomment-431162790 since 4.1.1 has been released?

**Comment by josephdickson** [comment]
@designsimply, not yet. I'm away from a computer traveling this week but hope to test it out this weekend and update this issue.

**Comment by josephdickson** [comment]
@designsimply and @chrisvanpatten and everyone else here.
I can confirm that the following removes the font slider and ability to manually set a font size.

`add_theme_support( 'disable-custom-font-sizes' ); `

Thank you everyone for helping me work through this issue. I really appreciate it.

Sorry for the late reply, I'm closing this as solved.
